aws-s3-transfer-manager/src/operation/upload/handle.rs

```diff
@@ -213,8 +213,12 @@ async fn complete_upload(handle: UploadHandle) -> Result<UploadOutput, crate::er
                 .set_sse_customer_key(handle.ctx.request.sse_customer_key.clone())
                 .set_sse_customer_key_md5(handle.ctx.request.sse_customer_key_md5.clone());
 
-            // check for user-provided full-object checksum...
             if let Some(checksum_strategy) = &handle.ctx.request.checksum_strategy {
+                // The Transfer Manager SEP recommends setting checksum-type on both
+                // CompleteMultipartUpload AND CreateMultipartUpload.
```

Comment:
you actually has to set it in both, otherwise, s3 will just error out.

Reply:
So weird...
If you don't pass the actual checksum value, then you only need to set type in CreateMPU.
But yeah, you're right, it's required in both if you're passing the actual value.

Reply:
removing snarky comments...

Reply:
> If the user provides a checksum for the object and the object is multipart uploaded then the Transfer Manager MUST set the x-amz-checksum-type header to "full_object" when calling CreateMultipartUpload and CompleteMultipartUpload

I guess it's only required to be set when user provides the checksum for full object. 

But, other cases, it's kind undefined.

I think it seems to work with S3 to set the type on complete MPU for other cases, but do we want to consider just move the type for complete MPU when user sets the full object checksum?

Reply:
I tested, and it's fine to set both. Tested the full matrix of Composite vs FullObject and with-checksum-value vs let-s3-calculate-it